WordPress/gutenberg
# Block Editor: Remove unused 'useIsDimensionsSupportValid' method

## What?
Removes the unused 'useIsDimensionsSupportValid' function.

P.S. We should stop using the hook naming convention for normal functions.

## Why?
The method hasn't been used since #48070 and has never been part of public API.

## Testing Instructions
 1. Open a Post or Page.
 2. Insert a Group block.
 3. Open Styles > Dimensions Panel
 4. Confirm there are no errors.

## Screenshots or screencast 

![CleanShot 2023-05-18 at 15 22 05](https://github.com/WordPress/gutenberg/assets/240569/f563e332-4059-4fd8-9d78-b8c274ca5ec0)
